### PR TITLE
Add crane 2.0 state-transfer enhancement doc

### PR DESCRIPTION
**Pull request by jmontleon**

### CODE REVIEW

File: enhancements/crane-2.0-state-transfer/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+This library will be focused on transfering state container within PVCs. We
+will not attempt to transfer state contained in:
+- ConfigMaps
+- Secrets
+- CustomResources
+
+## Proposal
+
+- Create a library with interfaces to enable transfer of data between clusters
+- The library will operate on one PVC at a time
```
Comment (shawn-hurley) [inline enhancements/crane-2.0-state-transfer/README.md:64]: why must it be one PVC at a time?

Comment (jmontleon) [reply enhancements/crane-2.0-state-transfer/README.md:64]: Maybe the wording is bad. The PVC is a unit to transfer, for which a server, and client is set up. I'm not talking about a case where we start up a server and client for a single PVC, transfer it, shut it down,  run the next...

Rather what you saw in the demo where we can launch several transfers in a namespace (or namespaces I suppose) and move data concurrently.

File: enhancements/crane-2.0-state-transfer/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+- The library will make it easy to run against all PVCs in one or more project
+- Because the library will need to speak to multiple clusters
```
[the remark this answers is not in the document]
Comment (jmontleon) [reply enhancements/crane-2.0-state-transfer/README.md:66]: Open to suggestions on implementing with two. In client code I see patterns on use --kubeconfig, KUBECONFIG env var or ~/.kube/.config. I didn't find an easy way to say use this file to create rest config 1, this file to create rest config 3, etc. I managed to do some hacky stuff for now to start working, but open to how to get this done better.

File: enhancements/crane-2.0-state-transfer/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  - KUBECONFIG / ~/.kube/config will always be used for destination
+  - Tools developed can build a config from another location, i.e. ~/.kube-src
+- Provide a CLI tool demonstrating how to implement functionality
```
[the remark this answers is not in the document]
Comment (jmontleon) [reply enhancements/crane-2.0-state-transfer/README.md:69]: Yes.  We planned on implementing the CLU in the crane repo, and the library in crane-lib, correct?
https://github.com/jmontleon/crane-lib

[the remark this answers is not in the document]
Comment (jmontleon) [reply enhancements/crane-2.0-state-transfer/README.md:69]: Yes.  We planned on implementing the CLU in the crane repo, and the library in crane-lib, correct?
https://github.com/konveyor/crane-lib
https://github.com/konveyor/crane

Library would be created in crane-lib, and then CLI in crane if it is done the way we discussed. I am focused on trying to implement the library here, not the CLI.

File: enhancements/crane-2.0-state-transfer/README.md (hunk 1 above)
Comment (jmontleon) [reply enhancements/crane-2.0-state-transfer/README.md:64]: Same response to https://github.com/konveyor/enhancements/pull/23#discussion_r626825764

Comment (jmontleon) [reply enhancements/crane-2.0-state-transfer/README.md:64]: In addition to all that, if we're building a reusable library, I think it makes sense to allow transferring individual PVCs. I want to set up a scheduled transfer of one PVC from one cluster to another (something like Scribe does), or backup two PVCs in a namespace, etc. Our goal of migrating every PVC in one NS isn't the only end game of a reusable library, I wouldn't think.

Comment (shawn-hurley) [reply enhancements/crane-2.0-state-transfer/README.md:64]: ack this makes sense. sounds like what we are saying is the interface for the library, will be single PVC's and the caller will be responsible for the concurrency or not.

File: enhancements/crane-2.0-state-transfer/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (shawn-hurley) [reply enhancements/crane-2.0-state-transfer/README.md:66]: For the library, can we say that when you initialize the thing that will transfer between two clusters will instead take two restconfigs. 

We can handle the CLI interface for this problem in a follow up

[the remark this answers is not in the document]
Comment (jmontleon) [reply enhancements/crane-2.0-state-transfer/README.md:66]: yes, makes sense.

File: enhancements/crane-2.0-state-transfer/README.md (hunk 1 above)
Comment (jmontleon) [reply enhancements/crane-2.0-state-transfer/README.md]: I updated this to try and make it clearer. If it still is not let me know and I'll take another stab at it.

File: enhancements/crane-2.0-state-transfer/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmontleon) [reply enhancements/crane-2.0-state-transfer/README.md]: Updated

**Review by shawn-hurley — APPROVED** [review]
I think this is ready to merge when you are ready @jmontleon

**Pull request merged**